src/semirings/string.ts

```diff
@@ -0,0 +1,103 @@
+import {Equals, Product, Union} from "../set";
+import {Semiring} from "../index";
+
+export class FormalLanguage {
+    readonly alphabet: Set<any>;
```

Comment:
Do you think we can use generics, ie `FormalLanguage<T>`; `Set<T>` instead of `Set<any>`?